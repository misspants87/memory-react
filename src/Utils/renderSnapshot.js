import React from 'react';
import renderer from 'react-test-renderer';

const renderSnapshot = (component) => {
  return renderer.create(component).toJSON();
}

export default renderSnapshot;
