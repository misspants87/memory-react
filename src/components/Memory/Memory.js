import React, { useState } from 'react';
import Game from '../Game/Game';

const Memory = () => {
  const [playAgain, setPlayAgain] = useState(false);

  const remountComponent = () => {
    setPlayAgain(!playAgain);
  }

  return (
    <Game
    data-test='Game Component'
    playAgain = {remountComponent.bind(this)}
    key={playAgain}/>
  )
}

export default Memory;
