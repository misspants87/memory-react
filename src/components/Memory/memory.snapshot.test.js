import React from 'react';
import Memory from './Memory';
import renderSnapshot from '../../Utils/renderSnapshot';

test('should render Memory Component correctly', () => {
  let snapshot = renderSnapshot(<Memory />);
  expect(snapshot).toMatchSnapshot();
})
