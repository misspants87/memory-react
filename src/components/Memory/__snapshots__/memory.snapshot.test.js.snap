// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Memory Component correctly 1`] = `
<div
  className="playerform-container"
>
  <h1
    className="title"
  >
    Memory
  </h1>
  <form
    className="form-container"
    data-test="Player Form"
    onSubmit={[Function]}
  >
    <label>
      Player 1:
      <input
        name="playerA"
        onChange={[Function]}
        placeholder="Enter name"
        type="text"
        value=""
      />
    </label>
    <label>
      Player 2:
      <input
        name="playerB"
        onChange={[Function]}
        placeholder="Enter name"
        type="text"
        value=""
      />
    </label>
    <button>
      Begin Game!
    </button>
  </form>
</div>
`;
